### PR TITLE
Check for essential FRUs

### PR DESCRIPTION
**Pull request by SunnySrivastava1984**
Some FRUs are marked as essential in the inventory JSON, which
implies that those FRUs should be present in the system at
power on.

The commit checks for presence of those FRUs when the system is
powered on.
In case any FRU in the list is found missing, appropriate
PEL is logged.

Test steps(rainier):
Check for present property for fru path
/xyz/openbmc_project/inventory/system/chassis/motherboard/base_op_panel_blyth

if it is false then proceed else set the property to false using
busctl set-property command.

execute obmcutil poweron
PEL should be logged for missing essential fru.

Signed-off-by: Sunny Srivastava <sunnsr25@in.ibm.com>

### CODE REVIEW

File: vpd-manager/manager.cpp [hunk 1]
```diff
@@ -657,6 +685,38 @@ void Manager::deleteFRUVPD(const sdbusplus::message::object_path path)
...
+    sd_bus* bus = nullptr;
+    sd_bus_default(&bus);
+
+    auto rc = sd_bus_call_method_async(
+        bus, NULL, loggerService, loggerObjectPath, loggerCreateInterface,
+        "Create",
+        [](sd_bus_message* msg, void* data, sd_bus_error*) {
+            (void)msg; // ignoring unused variable
+            (void)data;
+            std::cout << "Call back received for async bus call" << std::endl;
```
Comment (santoshpuranik) [inline vpd-manager/manager.cpp:708]: You will want to remove this

Comment (SunnySrivastava1984) [reply vpd-manager/manager.cpp]: yes, Done

File: vpd-manager/manager.cpp [hunk 2]
```diff
@@ -657,6 +685,38 @@ void Manager::deleteFRUVPD(const sdbusplus::message::object_path path)
...
+        pelSeverity = itr->second;
+    }
+
+    sd_bus* bus = nullptr;
+    sd_bus_default(&bus);
+
+    auto rc = sd_bus_call_method_async(
+        bus, NULL, loggerService, loggerObjectPath, loggerCreateInterface,
+        "Create",
+        [](sd_bus_message* msg, void* data, sd_bus_error*) {
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply vpd-manager/manager.cpp]: Removed callback as we don't need to perform any action on callback.

**Review by santoshpuranik — APPROVED** [review]

**Comment by SunnySrivastava1984** [comment]
Will fix SW553167 and SW550124

**Pull request merged**